### PR TITLE
[bug-fix] Fix Cookbook struct

### PR DESCRIPTION
**Pull request by afiune**
Match the `RootFiles` field to the right json map and add a new
`CookbookAccess` struct.

Closes https://github.com/go-chef/chef/issues/112

Signed-off-by: Salim Afiune <afiune@chef.io>

### CODE REVIEW

File: cookbook.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ type CookbookVersion struct {
 
 // CookbookMeta represents a Golang version of cookbook metadata
 type CookbookMeta struct {
-	Name            string                 `json:"cookbook_name,omitempty"`
+	Name            string                 `json:"name,omitempty"`
```
Comment (afiune) [inline cookbook.go]: I also fixed the `CookbookMeta` struct, here is an example of the `metadata` of a cookbook coming from a Chef Server:
```
  "metadata": {
    "name": "foo",
    "description": "Installs/Configures foo",
    "long_description": "Installs/Configures foo",
    "maintainer": "The Authors",
    "maintainer_email": "you@example.com",
    "license": "All Rights Reserved",
    "platforms": {},
    "dependencies": {},
    "recommendations": {},
    "suggestions": {},
    "conflicting": {},
    "providing": {
      "foo": ">= 0.0.0"
    },
    "replacing": {},
    "attributes": {},
    "groupings": {},
    "recipes": {
      "foo": ""
    },
    "version": "0.3.0"
  },
```

**Comment by afiune** [comment]
Hello there 👋  - I've got a quick bug fix for your convenience, I just recently added tests.

tenor-73535253

cc/ @MarkGibbons @spheromak

**Comment by afiune** [comment]
Hello there 👋  - I've got a quick bug fix for your convenience, I just recently added tests.

tenor-73535253

cc/ @MarkGibbons @spheromak

File: cookbook_test.go [hunk 2]
```diff
@@ -6,10 +6,67 @@ import (
 	"net/http"
 	//"os"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
 )
 
 const cookbookListResponseFile = "test/cookbooks_response.json"
-const cookbookTestFile = "test/cookbook.json"
```
Comment (afiune) [inline cookbook_test.go]: This constant wasn't in use so I assumed it was supposed to be used as a cookbook response. If it wasn't then I can create a new constant. 👍

**Comment by MarkGibbons** [comment]
Tagged as v0.0.2

File: cookbook.go [hunk 3]
```diff
@@ -73,8 +82,9 @@ type Cookbook struct {
 	Libraries    []CookbookItem `json:"libraries,omitempty"`
 	Providers    []CookbookItem `json:"providers,omitempty"`
 	Resources    []CookbookItem `json:"resources,omitempty"`
-	RootFiles    []CookbookItem `json:"templates,omitempty"`
+	RootFiles    []CookbookItem `json:"root_files,omitempty"`
 	Metadata     CookbookMeta   `json:"metadata,omitempty"`
+	Access       CookbookAccess `json:"access,omitempty"`
```
Comment (svagner) [inline cookbook.go:87]: @afiune, @MarkGibbons  could you please explain what does this `Access` field means https://docs.chef.io/server/api_chef_server/#get-23?

Comment (MarkGibbons) [reply cookbook.go:87]: @svagner Could you open an issue for this question?  I'll try to look over the weekend, but a comment on a closed PR is likely to get lost.  thanks

Comment (svagner) [reply cookbook.go:87]: Sure thing!

**Pull request merged**